### ISSUE TITLE
Octopilot exit code is "0" even when it fails to update repository

### ISSUE THREAD

**Issue by jozala** [dailymotion-oss/octopilot#110]
### Steps to reproduce:
1. Run Octopilot to update file in non-existing repository to cause failure
   ```
   octopilot \
      --repo "non-existing/example" \
      --update "regex(file=README.md,pattern='(?ms)(.*)')=replacing the content of the README.md file with this new content" \
      --log-level=debug
   ```
2. Check exit code
   ```
   echo $?
   ```

### Expected result
Exit code should be other than "0"  because Octopilot job failed replacing the content of the file

### Actual result
Exit code is "0" which indicates everything went well which is not true.

### Notes
This is especially a problem in CI/CD pipeline. Pipeline should fail when Octopilot cannot do its job, but it is passing, because of the exit code "0".

I have reproduced it inside Octopilot Docker container accessed with command:
```
docker run --name exittest --entrypoint=/bin/bash -it --rm ghcr.io/dailymotion-oss/octopilot:1.0.10
```

**Comment by vbehar** [comment]
hi, thanks for creating this issue. Yes indeed we have an open PR to support that: #84 but it was never finished/merged. I'll try to work on it tomorrow, or early next week

**Comment by vbehar** [comment]
implemented in #84 and released in https://github.com/dailymotion-oss/octopilot/releases/tag/v1.1.0
